src/Trakx.Kaiko.ApiClient.Tests/Integration/ReferenceDataTests.cs

```diff
@@ -51,4 +51,50 @@ public async Task InstrumentsClient_should_return_list_of_instruments()
         response.Result.Data.Should().NotBeNull();
         response.Result.Data.Should().HaveCountGreaterThan(0);
     }
+
+    [InlineData(false)]
+    [Theory]
+    public async Task Download_legacy_symbols(bool download)
+    {
+        if (!download)
+        {
+            download.Should().BeFalse();
+            return;
+        }
+
+        var client = ServiceProvider.GetRequiredService<IInstrumentsClient>();
+        var response = await client.GetAllInstrumentsAsync();
+
+        using var writer = new StreamWriter("kaiko-legacy-symbols.csv");
+
+        void Write(params string[] items)
```

Comment:
[suggestion] Maybe it's something that I'm used to, but I'd rather move this into a private static method, so the method's body would be cleaner.

Reply:
Method was moved. I tend to start writing local functions naturally, but I fully agree moving them to explicit methods outside of the main body is best.